hermes/probe/create/create.go

```diff
@@ -0,0 +1,186 @@
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+//
+// Author: Alicja Kwiecinska, GitHub: alicjakwie
+//
+// create contains all of the logic necessary to create the files with randomized contents in GCS
+package create
+
+import (
+	"context"
+	"crypto/sha1"
+	"fmt"
+	"io"
+	"math/rand"
+	"time"
+
+	"cloud.google.com/go/storage"
+	"github.com/google/cloudprober/logger"
+	"github.com/googleapis/google-cloud-go-testing/storage/stiface"
+	"github.com/googleinterns/step224-2020/hermes/probe"
+	"github.com/googleinterns/step224-2020/hermes/probe/metrics"
+
+	pb "github.com/googleinterns/step224-2020/hermes/proto"
+)
+
+const (
+	FileNamePrefixSize      = 9    // len("Hermes_ID")
+	begin                   = 1    // begin is the ID of the first HermesFile.
+	end                     = 50   // end is the ID of the last HermesFile.
+	maxFileSizeBytes        = 1000 // maxFileSizeBytes is the maximum allowed file size in bytes.
+	hermesAPILatencySeconds = "hermes_api_latency_s"
+)
+
+type randomFile struct {
+	id   int32 // id is a positive integer in the range [begin,end]
+	size int   // size is the file size in bytes.
+}
+
+type randomFileReader struct {
+	size int // size in bytes
+	i    int // currently reading this byte
+	rand *rand.Rand
+}
+
+// Read implements the standard Read interface, it reads data from a randomFileReader.
+// Argument:
+//      buf: a byte slice serves as a buffer
+// Returns:
+//	n:  the size of the buffer buf following a Read operation
+//	err: error it should be nil when not done reading and io.EOF once the whole file contents have been read
+func (r *randomFileReader) Read(buf []byte) (n int, err error) {
+	// check whether Read is done
+	if r.i >= r.size {
+		return 0, io.EOF
+	}
+	b := buf
+	if len(buf) > (r.size - r.i) {
+		// if the length of buffer is greater than the number of bytes left to read make the size of the buffer match the number of bytes left to read
+		b = buf[:(r.size - r.i)]
+	}
+	n, err = r.rand.Read(b) // n is now the length  of the buffer
+	if err != nil {
+		return n, err // in this case n = 0
+	}
+	r.i += n
+	return n, err
+}
+
+// NewReader creates a new randomFileReader and returns a pointer to it
+// Returns:
+//	*randomFileReader: a Reader that supports randomFiles. This Reader contains information about the file size in bytes as well as a pseudo-random bytes generator
+// Warning: The randomFileReader produced by NewReader is not thread safe.
+func (f *randomFile) NewReader() *randomFileReader {
+	//  id will serve as a Seed and i - index of the currently read byte  will be set to 0 automatically in the returned reader
+	return &randomFileReader{
+		size: f.size,
+		rand: rand.New(rand.NewSource(int64(f.id))), // rand.NewSource() requires an int64 as an argument
+	}
+}
+
+func newRandomFile(fileID int32, fileSize int) (*randomFile, error) {
```

Comment:
Why `int32`?

Reply:
I think I mentioned this in a reply to a previous comment (if not sorry) but it's because in State Journal which we're updating FileNames is a map[int32] string

Reply:
Sure, but why is *that* int32 in turn?
Is it because it's a proto field? If so, make it int64 and convert at the last moment. Use `int` everywhere in the code.

Reply:
@evanSpendlove made it map[int32] string so save space because it will be a max of 50 different types.
Should then fileID be an int and we should cast it to int32 and int64 in respective places?

Reply:
Using `int32` instead of `int` (which is typically `int64` on modern systems) doesn't save much space, especially if there are only that few entries in the map. It's a premature optimization. Just use `int` everwhere.

Reply:
Agreed. But since for the StateJournal is in a separate file, I've opened issue to fix this: https://github.com/googleinterns/step224-2020/issues/76.

@alicjakwie : can you add a TODO on this file to fix this after create.go is merged?

Reply:
added a TODO to change this (#76 )